scripts/prepdocs.py

```diff
@@ -1,623 +1,127 @@
-        ),
-        SimpleField(name="category", type="Edm.String", filterable=True, facetable=True),
-        SimpleField(name="sourcepage", type="Edm.String", filterable=True, facetable=True),
-        SimpleField(name="sourcefile", type="Edm.String", filterable=True, facetable=True),
-    ]
-    if args.useacls:
-        fields.append(
-            SimpleField(name="oids", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True)
+    use_vectors = not args.novectors
+    embeddings = None
+    if use_vectors and args.openaihost != "openai":
+        azure_open_ai_credential = credential if is_key_empty(args.openaikey) else AzureKeyCredential(args.openaikey)
+        embeddings = AzureOpenAIEmbeddingService(
+            open_ai_service=args.openaiservice,
+            open_ai_deployment=args.openaideployment,
+            open_ai_model_name=args.openaimodelname,
+            credential=azure_open_ai_credential,
+            disable_batch=args.disablebatchvectors,
+            verbose=args.verbose,
         )
-        fields.append(
-            SimpleField(name="groups", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True)
+    elif use_vectors:
+        embeddings = OpenAIEmbeddingService(
+            open_ai_model_name=args.openaimodelname,
+            credential=args.openaikey,
+            organization=args.openaiorg,
+            disable_batch=args.disablebatchvectors,
+            verbose=args.verbose,
         )
-
-    if args.index not in index_client.list_index_names():
-        index = SearchIndex(
-            name=args.index,
-            fields=fields,
-            semantic_settings=SemanticSettings(
-                configurations=[
-                    SemanticConfiguration(
-                        name="default",
-                        prioritized_fields=PrioritizedFields(
-                            title_field=None, prioritized_content_fields=[SemanticField(field_name="content")]
-                        ),
-                    )
-                ]
-            ),
-            vector_search=VectorSearch(
-                algorithm_configurations=[
-                    VectorSearchAlgorithmConfiguration(
-                        name="default", kind="hnsw", hnsw_parameters=HnswParameters(metric="cosine")
-                    )
-                ]
-            ),
+    else:
+        embeddings = None
+
+    print("Processing files...")
+    if args.datalakestorageaccount:
+        adls_gen2_creds = credential if is_key_empty(args.datalakekey) else args.datalakekey
+        print(f"Using Data Lake Gen2 Storage Account {args.datalakestorageaccount}")
+        list_file_strategy = ADLSGen2ListFileStrategy(
+            data_lake_storage_account=args.datalakestorageaccount,
+            data_lake_filesystem=args.datalakefilesystem,
+            data_lake_path=args.datalakepath,
+            credential=adls_gen2_creds,
+            verbose=args.verbose,
```

Comment:
There is no `verbose` argument in the ADLSGen2ListFileStrategy init method

Reply:
It's interesting mypy did not catch this, will fix